src/jit/codegenarm64.cpp

```diff
@@ -4898,6 +4898,10 @@ void CodeGen::genStoreLclTypeSIMD12(GenTree* treeNode)
 
     // 4-byte write
     getEmitter()->emitIns_S_R(INS_str, EA_4BYTE, tmpReg, varNum, offs + 8);
+
+    genUpdateLife(treeNode);
+
+    compiler->lvaTable[varNum].lvRegNum = REG_STK;
```

Comment:
I don't understand why this is required. We shouldn't be calling this method unless we have a field - either a `STORE_LCL_FLD` or a `STORE_LCL_VAR` of a "dependently promoted" field (i.e. a field that was promoted to a lclVar, but then for some reason was forced to remain part of the parent struct on the stack). In either of those cases, we should never have anything but `lvRegNum == REG_STK`. Unless, of course, there's something I'm missing here.

Reply:
It may not be required for SIMD, but it was observed that SIMD16, SIMD8, and SIMD32 calls these functions in the normal paths in `genCodeForStoreLclVar()` and  `genCodeForStoreLclFld()`.

If it is not needed I'll close the issue.